sonar-php-plugin/src/main/java/org/sonar/plugins/php/PHPSensor.java

```diff
@@ -190,34 +221,35 @@ private void saveSyntaxHighlighting(List<SyntaxHighlightingData> highlightingDat
 
       highlightingBuilder.done();
     }
-
   }
 
   @Nullable
-  <P extends Perspective<?>> P perspective(Class<P> clazz, InputFile file) {
+  <P extends Perspective> P perspective(Class<P> clazz, InputFile file) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "Perspective"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)